WalletWasabi.Tests/UnitTests/Clients/WhaleCoinjoinTests.cs

```diff
@@ -0,0 +1,229 @@
+		for (int i = 0; i < nbOtherClients; i++)
+		{
+			KeyManager kmTmp = KeyManager.CreateNew(out _, "", Network.RegTest);
+			otherHdPub[i] = BitcoinFactory.CreateHdPubKey(kmTmp);
+		}
+
+		var whaleSmartCoins = new List<SmartCoin> { BitcoinFactory.CreateSmartCoin(hdPub, (decimal)whaleAmountBtc) };
+		var totalVSizeWhale = 0;
+		var counter = 0;
+		while (whaleMinInputAnonSet < anonScoreTarget && counter <= maxTestRounds)
+		{
+			counter++;
+			var whaleSelectedSmartCoins = SelectCoinsForRound(whaleSmartCoins, anonScoreTarget, roundParams, mockSecureRandom);
+			var whaleSelectedCoins = whaleSelectedSmartCoins.Select(sm => sm.Coin);
+			if (!whaleSelectedCoins.Any())
+			{
+				break;
+			}
+
+			whaleSmartCoins.Remove(whaleSelectedSmartCoins);
+
+			var otherSmartCoins = CreateOtherSmartCoins(mockSecureRandom, otherHdPub, otherNbInputsPerClient, otherClientsAmountMin, otherClientsAmountMax);
+			var otherSelectedSmartCoins = new List<List<SmartCoin>>();
+			var otherSelectedCoins = new List<IEnumerable<Coin>>();
+			var otherOutputs = new List<List<(Money, int)>>();
+
+			foreach (var other in otherSmartCoins)
+			{
+				var tmpSelectedSmartCoins = SelectCoinsForRound(other, anonScoreTarget, roundParams, mockSecureRandom);
+				otherSelectedSmartCoins.Add((tmpSelectedSmartCoins.ToList()));
+				otherSelectedCoins.Add(tmpSelectedSmartCoins.Select(x => x.Coin));
+			}
+
+			foreach (var otherSelectedCoin in otherSelectedCoins)
+			{
+				otherOutputs.Add(DecomposeWithAnonSet(otherSelectedCoin, otherSelectedCoins.SelectMany(x => x).Except(otherSelectedCoin).Concat(whaleSelectedCoins), mockSecureRandom));
+			}
+
+			var whaleInputs = whaleSelectedSmartCoins.Select(x => (x.Amount, (int)x.HdPubKey.AnonymitySet));
+			var whaleOutputs = DecomposeWithAnonSet(whaleSelectedCoins, otherSelectedCoins.SelectMany(x => x), mockSecureRandom);
+			var tx = BitcoinFactory.CreateSmartTransaction(otherSelectedCoins.SelectMany(x => x).Count(), otherOutputs.SelectMany(x => x).Select(x => x.Item1), whaleInputs, whaleOutputs);
+			analyser.Analyze(tx);
+
+			whaleSmartCoins.Add(tx.WalletOutputs.ToList());
+			whaleMinInputAnonSet = whaleSmartCoins.Min(x => x.HdPubKey.AnonymitySet);
+			totalVSizeWhale += tx.WalletInputs.Sum(x => x.ScriptPubKey.EstimateInputVsize());
+
+			if (counter % displayWhaleCoinsEachRounds == 0)
+			{
+				DisplayWhaleCoinsAnonSet(counter, totalVSizeWhale, whaleSmartCoins);
+			}
+		}
+
+		WriteLine($"whaleAmountBtc: {whaleAmountBtc}, otherClientsAmountMin: {otherClientsAmountMin}, otherClientsAmountMax: {otherClientsAmountMax}, randomSeed: {randomSeed}");
+		WriteLine(counter >= maxTestRounds
+			? $"FAILED whaleMinInputAnonSet only reached {whaleMinInputAnonSet} after {counter} rounds"
+			: $"PASSED after {counter} rounds -> Total vSize Whale: {totalVSizeWhale}");
+	}
+
+	public class TestRandomSeed : SecureRandom
+	{
+		private readonly Random _seededInstance;
+		public int CurrentDepth = 0;
```

Comment:
The data is never read from this prop. Also naming rule violation.

Reply:
This is to test random inconsistencies. I removed it.